### DISCUSSION TITLE
Quarkus @ConfigProperty does not inject value

### DISCUSSION THREAD

**Discussion by naamabedev2** [quarkusio/quarkus#40128] [Q&A]
I am using quarkus @ConfigProperty annotation in order to inject value to a String data member:

```
@ApplicationScoped
public class ActiveMQRout extends RouteBuilder {

    public String getOrderMessage() {
        return orderMessage;
    }
    public void setOrderMessage(String orderMessage) {
        this.orderMessage = orderMessage;
    }
    
    @ConfigProperty(name="order")
    String orderMessage;

//more code that uses orderMessage
}
```
In application.properties file I have this value: order=order2

the value of orderMessage stay null. I have tried using the following option instead of injection:

orderMessage = ConfigProvider.getConfig().getValue("order", String.class);

and it is working. how can I make this work with injection (by using @ConfigProperty annotation)?

(I have tried using the @Singleton annotation for the class ActiveMQRout but still did not work. I also added getter and setter but still did not work)

[a comment by an automated account is not shown]

**Comment by radcortez** [comment]
Hi @naamabedev2. Can you please provide us with a reproducer?

Config injection works for CDI Beans. Is `ActiveMQRout` used as a CDI Bean? Having annotations has no effect if the class constructor is called explicitly.
